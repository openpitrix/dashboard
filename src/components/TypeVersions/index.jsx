import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { observer } from 'mobx-react';
import classnames from 'classnames';
import { translate } from 'react-i18next';

import { getVersionTypesName } from 'config/version-types';
import _ from 'lodash';

import styles from './index.scss';

@translate()
@observer
export default class TypeVersions extends Component {
  static propTypes = {
    activeType: PropTypes.string,
    activeVersion: PropTypes.string,
    changeType: PropTypes.func,
    className: PropTypes.string,
    types: PropTypes.array,
    versions: PropTypes.array
  };

  static defaultProps = {
    activeType: '',
    activeVersion: '',
    changeType: _.noop,
    types: [],
    versions: []
  };

  changeType(value, type) {
    this.props.changeType(value, type);
  }

  render() {
    const {
      className,
      types,
      versions,
      activeType,
      activeVersion,
      t
    } = this.props;

    return (
      <div className={classnames(styles.typeVersions, className)}>
        <dl>
          <dt>{t('Delivery type')}:</dt>
          <dd className={styles.types}>
            {types.map(type => (
              <label
                key={type}
                onClick={() => this.changeType(type, 'activeType')}
                className={classnames({
                  [styles.active]: (activeType || types[0]) === type
                })}
              >
                {getVersionTypesName(type) || t('None')}
              </label>
            ))}
          </dd>
        </dl>
        <dl>
          <dt>{t('Version No')}:</dt>
          <dd className={styles.types}>
            {versions.map(item => (
              <label
                key={item.version_id}
                onClick={() => this.changeType(item.version_id, 'activeVersion')
                }
                className={classnames({
                  [styles.active]: activeVersion === item.version_id
                })}
              >
                {item.name || t('None')}
              </label>
            ))}
          </dd>
        </dl>
      </div>
    );
  }
}
