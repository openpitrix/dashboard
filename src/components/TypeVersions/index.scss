@import '~scss/vars';

.typeVersions {
  dl {
    margin-bottom: 20px;
    overflow: hidden;
  }

  dt {
    @include normal-font;
    display: inline-block;
    width: 72px;
    padding: 6px 12px 6px 0;
    text-align: right;
  }

  dd {
    display: inline-block;
    float: right;
    width: calc(100% - 85px);
  }

  .types {
    // margin-bottom: 16px;

    > label {
      display: inline-block;
      padding: 8px 12px;
      background-color: $N0;
      border: 1px solid $N10;
      font-size: 14px;
      line-height: 14px;
      color: $N300;
      cursor: pointer;

      &:first-child {
        border-top-left-radius: 2px;
        border-bottom-left-radius: 2px;
      }

      &:last-child {
        border-top-right-radius: 2px;
        border-bottom-right-radius: 2px;
      }

      &.active {
        color: $P75;
        border-color: $P75;
      }
    }
  }
}
