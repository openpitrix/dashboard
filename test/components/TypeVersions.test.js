import TypeVersions from 'components/TypeVersions';
import { mount } from 'enzyme';

const props = {
  types: ['helm'],
  versions: [{
    version_id: 'appr-132',
    name: '1.0.0'
  }]
};
describe('Component TypeVersions', () => {
  it('basic render', () => {
    const wrapper = render(<TypeVersions {...props} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('call onClick', () => {
    const mockClick = jest.fn();
    const wrapper = mount(
      <TypeVersions {...props} changeType={mockClick}>
        Click me
      </TypeVersions>
    );
    wrapper.simulate('click');
  });
});
