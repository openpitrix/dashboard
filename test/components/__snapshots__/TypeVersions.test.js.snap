// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component TypeVersions basic render 1`] = `
<div
  class="typeVersions"
>
  <dl>
    <dt>
      Delivery type:
    </dt>
    <dd>
      <div
        class="types"
      >
        <label
          class="active"
        >
          Helm
        </label>
      </div>
    </dd>
  </dl>
  <dl>
    <dt>
      Version No:
    </dt>
    <dd>
      <div
        class="types"
      >
        <label
          class=""
        >
          1.0.0
        </label>
      </div>
    </dd>
  </dl>
</div>
`;
